### DISCUSSION TITLE
Anyway to implement a custom new compiled version of chromium?

### DISCUSSION THREAD

**Discussion by MichaelEpicA** [Voltstro-Studios/UnityWebBrowser#111] [General]
Wanting to test something out, and I can't really figure out a way to do it.

**Comment by Voltstro** [comment]
You would need to recompile libcef targeting the same **exact** version that CefGlue expects (version can be found in `src/ThirdParty/CefGlue/CefGlue/Interop/version.g.cs`). Once you have done that you can just replace the binaries in `src/ThirdParty/Libs/cef/<OS>`, and rebuild the CEF engine.

**Reply by MichaelEpicA** [reply]
So basically, get the version it wants, replace the binaries in the libs, then rebuild UnityWebBrowser? Also using github, I can't seem to click on the CefGlue Stuff i can only get into libs

**Reply by Voltstro** [reply]
Yes.

You can't view the CefGlue stuff on GitHub web view because its a submodule thats hosted on GitLab.

**Reply by MichaelEpicA** [reply]
Huh ok, this is a problem, I tried to build using the automate.py thing and I just got 'the minimum supported branch is 3071."
NVM IM STUPID
But yeah im worried because last time it just got stuck on working on src, and when I pressed enter since the time wasnt moving it just stopped

**Reply by MichaelEpicA** [reply]
Trying to compile, and I keep getting this error: ERROR at //skia/BUILD.gn:395:7: Item not found
      "//third_party/skia/src/utils/SkThreadUtils_pthread.cpp",
      ^-------------------------------------------------------
You were trying to remove "//third_party/skia/src/utils/SkThreadUtils_pthread.cpp"
from the list but it wasn't there.
See //BUILD.gn:94:5: which caused the file to be included.
    "//skia:skia_unittests",
    ^----------------------
Traceback (most recent call last):
  File "E:\Build_Chromium\code\chromium_git\chromium\src\cef\tools\gclient_hook.py", line 146, in <module>
    RunAction(src_dir, cmd)
  File "E:\Build_Chromium\code\chromium_git\chromium\src\cef\tools\gclient_util.py", line 35, in RunAction
    gclient_utils.CheckCallAndFilter(
  File "E:\Build_Chromium\code\depot_tools\gclient_utils.py", line 715, in CheckCallAndFilter
    raise subprocess2.CalledProcessError(
subprocess2.CalledProcessError: Command 'gn gen out\\Debug_GN_x64 --ide=vs2019 --sln=cef --filters=//cef/*' returned non-zero exit status 1 in E:\Build_Chromium\code\chromium_git\chromium\src
ERROR at //skia/BUILD.gn:395:7: Item not found
      "//third_party/skia/src/utils/SkThreadUtils_pthread.cpp",
      ^-------------------------------------------------------
You were trying to remove "//third_party/skia/src/utils/SkThreadUtils_pthread.cpp"
from the list but it wasn't there.
See //BUILD.gn:94:5: which caused the file to be included.
    "//skia:skia_unittests",
    ^----------------------

**Reply by Voltstro** [reply]
You'd be better off asking the CEF devs, as compiling CEF has nothing to do with this project.

**Comment by MichaelEpicA** [comment]
On Mon, Sep 12, 2022 at 9:20 AM Voltstro ***@***.***> wrote:

> You'd be better off asking the CEF devs, as compiling CEF has nothing to
> do with this project.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Voltstro-Studios/UnityWebBrowser/discussions/111#discussioncomment-3626579>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AH5B5CMOY23TMROI67ALJE3V5432ZANCNFSM6AAAAAAQJAFCDM>
> .
> You are receiving this because you authored the thread.Message ID:
> <Voltstro-Studios/UnityWebBrowser/repo-discussions/111/comments/3626579@
> github.com>
>
Ah alright
